holoviews/core/operation.py

```diff
@@ -64,6 +70,50 @@ def get_overlay_bounds(cls, overlay):
             raise ValueError("Extents across the overlay are inconsistent")
 
 
+class DynamicOperation(Operation):
+    """
+    Dynamically applies a function to the elements of a HoloMap
+    or DynamicMap. Will return a DynamicMap wrapping the original
+    map object, which will lazily evaluate when a key is requested.
+    """
+
+    def __call__(self, map_obj, function, **kwargs):
+        callback = self._dynamic_operation(map_obj, function, **kwargs)
```

Comment:
I would prefer it if ``__call__`` has the same signature as ``Operation`` and if instead of passing in ``function`` as an argument you use the ``_process`` method. This would make ``DynamicOperation`` more consistent with ``Operation`` and you could always create a very generic ``DynamicOperation`` subclass that takes in a function from the outside to achieve the same functionality.